backends/kv/writer.go

```diff
@@ -425,3 +435,16 @@ func (a *Appender) validateSchema(frame frames.Frame) error {
 
 	return nil
 }
+
+// Check if the current error was caused specifically because the condition was evaluated to false.
+func isFalseConditionError(err error) bool {
+	errString := err.Error()
+
+	if strings.Count(errString, errorCodeString) == 2 &&
+		strings.Contains(errString, falseConditionOuterErrorCode) &&
+		strings.Contains(errString, falseConditionInnerErrorCode) {
```

Comment:
You can use `errors.Cause(err)`.

Reply:
Not sure that will help since it's just one error with a "big" string like the one in the description.

Reply:
`err` is actually a struct. `err.Error()` returns one string.

Reply:
Will leave as is for now.